pkgs/applications/editors/bluej/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, makeWrapper, jetbrains }:
+
+stdenv.mkDerivation rec {
+  pname = "bluej";
+  version = "4.2.2";
+  src = fetchurl {
+    url = "https://www.bluej.org/download/files/BlueJ-linux-${builtins.replaceStrings ["."] [""] version}.deb";
```

Comment:
You should have easier life with this url:

https://www.bluej.org/download/files/BlueJ-generic-422.jar

From here: https://www.bluej.org/versions.html .

Reply:
You might think that, but it's not actually the case. That jar is an installer, which you need to run once, input some things into a graphical application and which will then install BlueJ itself into the paths you specified in that graphical installer. (See https://bluej.org/generic-installation-instructions.html).

Reply:
OK :)

Reply:
It may worth writing this in a comment.